project-koku/nise-populator
# COST-2990: add oci to nise populator flow

Adding OCI  to nise populator flow.

## Testing Instructions:

- Add env vars:
  You'll need the following OCI config env variables in you `.env` file.
  You can refer to this [oracle doc](https://docs.oracle.com/en-us/iaas/Content/API/Concepts/apisigningkey.htm#Required_Keys_and_OCIDs) for all required keys.

   ```
   OCI_CLI_USER=ocid1.user.oc1..<unique_ID>
   OCI_CLI_FINGERPRINT=<your_fingerprint>
   OCI_CREDENTIALS=<private_your_key>
   OCI_CLI_TENANCY=ocid1.tenancy.oc1..<unique_ID>
   OCI_REGION=<oci_region>
   OCI_BUCKET_NAME=<oci_bucket_name>
   OCI_NAMESPACE=<oci_namespace>  # bucket namespace
   ```

- To quickly test OCI only or don't have info for the other providers, you can [temporarily comment this line out](https://github.com/project-koku/nise-populator/blob/main/nise-populator/sources/source_factory.py#L18) and add, `self.valid_sources = [OCI]`

- Run app.py
   `python3 nise-populator/app.py`
You should see something like this in the logs:

```
2022-09-12 17:47:34,022 - nise.util.log - INFO - File reports/cost-csv/report_cost-0001_2022-09.csv.gz uploaded to OCI Storage oci-test-bucket bucket.
```
